Tasks/Common/webdeployment-common/azurerestutility.ts

```diff
@@ -213,3 +213,95 @@ export async function getAzureRMWebAppConfigDetails(SPN, webAppName: string, res
     });
     return deferred.promise;
 }
+
+export async function getWebAppAppSettings(SPN, webAppName: string, resourceGroupName: string, deployToSlotFlag: boolean, slotName: string, obj: Object)
+{
+	if(!deployToSlotFlag) {
+       var requestURL = armUrl + 'subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType EQ \'Microsoft.Web/Sites\' AND name EQ \'' + 
+                          webAppName + '\'&api-version=2016-07-01';
+        var accessToken = await getAuthorizationToken(SPN);
+        var headers = {
+            authorization: 'Bearer '+ accessToken
+        };
+        var webAppID = await getAzureRMWebAppID(SPN, webAppName, requestURL, headers);
+        tl.debug('Web App details : ' + webAppID.id);
+        resourceGroupName = webAppID.id.split ('/')[4];
+        tl.debug('AzureRM Resource Group Name : ' + resourceGroupName);
+    }
+
+    var deferred = Q.defer<any>();
+	
+	var accessToken = await getAuthorizationToken(SPN);
```

Comment:
I see multiple calls to getAuthorizationToken within this same function. Do we have to do it multiple times? Can't we get once and use it for multiple calls?